pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -99,43 +99,52 @@ def get_desc(self):
         - descriptorOffset: Length of the C1:WF ALL,#9000000346 message
 
         """
-        command = "C{ch}:WF? DESC"
+        if descriptor is None:
+            command = "C{ch}:WF? DESC"
+            self.write(command)
+            descriptor = self.read_bytes(count=-1, break_on_termchar=True)
+        command = "WFSU?"
         self.write(command)
-        descriptor = self.read_bytes(count=-1, break_on_termchar=True)
+        wfsu = self.read()
+        sparsing = int(wfsu.split(",")[1])
+        number = int(wfsu.split(",")[3])
+        first = int(wfsu.split(",")[5])
         descriptorOffset = 21
-        (numDataPoints,) = struct.unpack_from(
-            "l",
-            descriptor,
-            offset=descriptorOffset + 60,
-        )
-        (verticalGain,) = struct.unpack_from(
-            "f",
-            descriptor,
-            offset=descriptorOffset + 156,
-        )
-        (verticalOffset,) = struct.unpack_from(
-            "f",
-            descriptor,
-            offset=descriptorOffset + 160,
-        )
-        (horizInterval,) = struct.unpack_from(
-            "f",
-            descriptor,
-            offset=descriptorOffset + 176,
-        )
-        (horizOffset,) = struct.unpack_from(
-            "d",
-            descriptor,
-            offset=descriptorOffset + 180,
-        )
-        descriptorDictionnary = {
-            "numDataPoints": numDataPoints,
-            "verticalGain": verticalGain,
-            "verticalOffset": verticalOffset,
-            "horizInterval": horizInterval,
-            "horizOffset": horizOffset,
-            "descriptorOffset": descriptorOffset,
-        }
+        descriptorDictionnary = {"sparsing": sparsing, "numDataPoints": number, "firstPoint": first}
```

Comment:
```suggestion
        descriptor_dictionary = {"sparsing": sparsing, "numDataPoints": number, "firstPoint": first}
```
First, the word _dictionary_ does not contain two times _n_.
Second, python recommendation is to use an underscore to separate words in a variable (or method) name.
(btw. Classes use CamelCase)